videos/threeplay_api.py

```diff
@@ -0,0 +1,129 @@
+    url = "https://api.3playmedia.com/v3/files"
+    response = requests.get(url, payload)
+    if response:
+        return response.json()
+    else:
+        return {}
+
+
+def threeplay_remove_tags(threeplay_video_id):
+    """3play patch to remove tag from video file"""
+    payload = {"label": "", "api_key": settings.THREEPLAY_API_KEY}
+    url = f"https://api.3playmedia.com/v3/files/{threeplay_video_id}"
+    requests.patch(url, payload)
+
+
+def threeplay_transcript_api_request(youtube_id):
+    """3play data requst to get transcripts by youtube_id"""
+    payload = {
+        "media_file_reference_id": youtube_id,
+        "api_key": settings.THREEPLAY_API_KEY,
+    }
+    url = "https://api.3playmedia.com/v3/transcripts"
+    response = requests.get(url, payload)
+    if response:
+        return response.json()
+    else:
+        return {}
+
+
+def upload_to_s3(source_url, s3_destination):
+    """Upload transcripts to s3"""
+
+    request = Request(source_url)
+
+    try:
+        response = urlopen(request)
+    except HTTPError:
+        log.error(
+            "Could not open 3play transcript at %s",
+            source_url,
+        )
+        return False
+
+    s3 = boto3.resource("s3")
+    bucket_name = settings.AWS_STORAGE_BUCKET_NAME
+    bucket = s3.Bucket(bucket_name)
+
+    bucket.upload_fileobj(
+        Fileobj=response, Key=s3_destination, ExtraArgs={"ACL": "public-read"}
+    )
+
+    return True
+
+
+def update_transcripts_for_video(video):
+    """Download transcripts from 3play, upload them to s3 and update the Video file"""
+
+    youtube_video_file = video.videofiles.filter(
+        destination=DESTINATION_YOUTUBE
+    ).first()
+
+    if not (youtube_video_file and youtube_video_file.destination_id):
+        return
+    else:
+        youtube_id = youtube_video_file.destination_id
+
+    threeplay_transcript_json = threeplay_transcript_api_request(youtube_id)
+
+    if (
+        threeplay_transcript_json.get("data")
+        and len(threeplay_transcript_json.get("data")) > 0
+    ):
+        transcript_id = threeplay_transcript_json["data"][0].get("id")
+        media_file_id = threeplay_transcript_json["data"][0].get("media_file_id")
+
+        s3_base = (
+            f"{settings.S3_TRANSCRIPTS_PREFIX}/{video.website.short_id}/{youtube_id}/"
+        )
+
+        transcript_url_base = (
+            f"https://api.3playmedia.com/p/files/{media_file_id}/threeplay_transcripts/"
+            f"{transcript_id}?project_id={settings.THREEPLAY_PROJECT_ID}"
+        )
+
+        pdf_url = transcript_url_base + "&format_id=46"
+
+        pdf_s3_destination = s3_base + "transcript.pdf"
+
+        webvtt_url = transcript_url_base + "&format_id=51"
+
+        webvtt_s3_destination = s3_base + "transcript_webvtt"
+
+        if upload_to_s3(pdf_url, pdf_s3_destination):
+            video.pdf_transcript_s3_url = (
+                f"s3://{settings.AWS_STORAGE_BUCKET_NAME}/" + pdf_s3_destination
```

Comment:
Should the urls be in a format that hugo can use to make a downloadable html link, instead of an s3-specific link?  Actually I guess hugo could figure that out from the above format anyway, so this might be fine.